scripts/build_backend.py

```diff
@@ -18,116 +18,132 @@
+
+    backends = os.path.join(dpctl_dir, "dpctl-capi")
+
+    ENABLE_LO_PROGRAM_CREATION = "ON" if l0_support else "OFF"
+
+    if IS_LIN:
+        if CODE_COVERAGE:
+            cmake_args = [
+                "cmake",
+                "-DCMAKE_BUILD_TYPE=Debug",
+                "-DCMAKE_INSTALL_PREFIX=" + INSTALL_PREFIX,
+                "-DCMAKE_PREFIX_PATH=" + INSTALL_PREFIX,
+                "-DCMAKE_C_COMPILER:PATH="
+                + os.path.join(DPCPP_ROOT, "bin", "clang"),
+                "-DCMAKE_CXX_COMPILER:PATH="
+                + os.path.join(DPCPP_ROOT, "bin", "dpcpp"),
+                "-DDPCTL_ENABLE_LO_PROGRAM_CREATION="
+                + ENABLE_LO_PROGRAM_CREATION,
+                "-DDPCTL_BUILD_CAPI_TESTS=ON",
+                "-DDPCTL_GENERATE_COVERAGE=ON",
+                "-DDPCTL_COVERAGE_REPORT_OUTPUT_DIR=" + dpctl_dir,
+                backends,
+            ]
+            subprocess.check_call(
+                cmake_args, stderr=subprocess.STDOUT, shell=False
+            )
+            subprocess.check_call(["make", "V=1", "-j", "4"])
+            subprocess.check_call(["make", "install"])
+            subprocess.check_call(["make", "lcov-genhtml"])
+        else:
+            cmake_args = [
+                "cmake",
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_INSTALL_PREFIX=" + INSTALL_PREFIX,
+                "-DCMAKE_C_COMPILER:PATH="
+                + os.path.join(DPCPP_ROOT, "bin", "clang"),
+                "-DCMAKE_CXX_COMPILER:PATH="
+                + os.path.join(DPCPP_ROOT, "bin", "dpcpp"),
+                "-DCMAKE_PREFIX_PATH=" + INSTALL_PREFIX,
+                "-DDPCTL_ENABLE_LO_PROGRAM_CREATION="
+                + ENABLE_LO_PROGRAM_CREATION,
+                backends,
+            ]
+            subprocess.check_call(
+                cmake_args, stderr=subprocess.STDOUT, shell=False
+            )
+            subprocess.check_call(["make", "V=1", "-j", "4"])
+            subprocess.check_call(["make", "install"])
+
+        os.chdir(dpctl_dir)
+        for file in glob.glob(
+            os.path.join(dpctl_dir, "install", "lib", "*.so")
+        ):
+            shutil.copy(file, os.path.join(dpctl_dir, "dpctl"))
+    elif IS_WIN:
         cmake_args = [
             "cmake",
+            "-G",
+            "Ninja",
             "-DCMAKE_BUILD_TYPE=Release",
             "-DCMAKE_INSTALL_PREFIX=" + INSTALL_PREFIX,
-            "-DCMAKE_PREFIX_PATH=" + INSTALL_PREFIX,
-            "-DDPCPP_INSTALL_DIR=" + DPCPP_ROOT,
             "-DCMAKE_C_COMPILER:PATH="
-            + os.path.join(DPCPP_ROOT, "bin", "clang"),
+            + os.path.join(DPCPP_ROOT, "bin", "clang-cl.exe"),
             "-DCMAKE_CXX_COMPILER:PATH="
-            + os.path.join(DPCPP_ROOT, "bin", "dpcpp"),
-            "-DDPCTL_ENABLE_LO_PROGRAM_CREATION=ON",
+            + os.path.join(DPCPP_ROOT, "bin", "dpcpp.exe"),
+            "-DCMAKE_PREFIX_PATH=" + INSTALL_PREFIX,
```

Comment:
Should L0 be enabled to be enabled on Windows too?

Reply:
#386 is specifically attempting to do so.